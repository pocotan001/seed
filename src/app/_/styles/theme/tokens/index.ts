export * from "./aspect-ratio";
export * from "./border-width";
export * from "./colors";
export * from "./font-sizes";
export * from "./fonts";
export * from "./grid-column";
export * from "./grid-template-columns";
export * from "./media";
export * from "./radii";
export * from "./sizes";
export * from "./space";
export * from "./z-indices";
