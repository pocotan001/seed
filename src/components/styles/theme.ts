export type ITheme = typeof theme;

const theme = {
  colors: {
    white: "#fff",
    black: "#000",
    // Grey
    grey50: "#fafafa",
    grey100: "#f5f5f5",
    grey200: "#eeeeee",
    grey300: "#e0e0e0",
    grey400: "#bdbdbd",
    grey500: "#9e9e9e",
    grey600: "#757575",
    grey700: "#616161",
    grey800: "#424242",
    grey900: "#212121",
    // Pink
    pink50: "#fce4ec",
    pink100: "#f8bbd0",
    pink200: "#f48fb1",
    pink300: "#f06292",
    pink400: "#ec407a",
    pink500: "#e91e63",
    pink600: "#d81b60",
    pink700: "#c2185b",
    pink800: "#ad1457",
    pink900: "#880e4f"
  },

  fonts: {
    sans: '"Helvetica Neue", Helvetica, Arial, sans-serif'
  },

  breakpoints: {
    tablet: 768,
    phone: 376
  },

  dimensions: {
    sidebarWidth: 280
  },

  zIndex: {
    loading: 100
  }
};

export default theme;
