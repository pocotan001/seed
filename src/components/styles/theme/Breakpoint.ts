enum Breakpoint {
  tablet = 768,
  phone = 376
}

export default Breakpoint;
