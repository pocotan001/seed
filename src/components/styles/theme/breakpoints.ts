export type IBreakpointKey = keyof typeof breakpoints;

const breakpoints = Object.freeze({
  tablet: 768,
  phone: 376
});

export default breakpoints;
