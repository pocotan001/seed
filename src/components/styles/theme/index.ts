export { default as breakpoints } from "./breakpoints";
export { default as colors } from "./colors";
export { default as dimensions } from "./dimensions";
export { default as fonts } from "./fonts";
export { default as zIndex } from "./zIndex";
