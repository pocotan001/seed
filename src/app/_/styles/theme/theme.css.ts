import { createGlobalTheme, createVar } from "@vanilla-extract/css";
import * as tokens from "./tokens";

export type CSSVarFunction = ReturnType<typeof createVar>;
export const vars = createGlobalTheme(":root", tokens);
