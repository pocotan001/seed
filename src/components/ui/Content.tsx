import styled from "styled-components";
import withMargin, {
  MarginProps
} from "~/components/styles/extends/withMargin";
import { Color } from "~/components/styles/theme";

interface ContentProps extends MarginProps {
  children?: React.ReactNode;
}

const Content = styled<ContentProps, "div">("div")`
  > *:first-child {
    margin-top: 0 !important;
  }

  > *:last-child {
    margin-bottom: 0 !important;
  }

  blockquote,
  dl,
  ol,
  p,
  pre,
  table,
  ul {
    margin: 16px 0;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: normal;
    margin: 24px 0;
    color: ${Color.Grey800};
  }

  h1 {
    font-size: 2rem;
  }

  h2 {
    font-size: 1.75rem;
  }

  h3 {
    font-size: 1.5rem;
  }

  h4 {
    font-size: 1.25rem;
  }

  h5 {
    font-size: 1.125rem;
  }

  h6 {
    font-size: 1rem;
  }

  a {
    text-decoration: none;
    color: ${Color.Pink300};

    &:hover {
      text-decoration: underline;
    }
  }

  blockquote {
    padding: 0 16px;
    border-left: 4px solid ${Color.Grey300};
  }

  hr {
    height: 4px;
    border: 0;
    background: ${Color.Grey300};
  }

  ol {
    margin-left: 2em;
    list-style: decimal outside;
  }

  ul {
    margin-left: 2em;
    list-style: disc outside;
  }

  ul ul {
    margin-top: 8px;
    list-style-type: circle;
  }

  li + li {
    margin-top: 8px;
  }

  li p {
    margin: 8px 0;
  }

  img {
    max-width: 100%;
    vertical-align: middle;
  }

  code {
    margin: 0 4px;
    padding: 0 4px;
    white-space: nowrap;
    border: 1px solid ${Color.Grey300};
    border-radius: 3px;
    background: ${Color.Grey100};
  }

  pre {
    overflow: auto;
    padding: 16px;
    background: ${Color.Grey100};
    -webkit-overflow-scrolling: touch;
  }

  pre code {
    white-space: pre;
    border: 0;
    background: transparent;
    margin: 0;
  }

  table {
    width: 100%;
  }

  table td,
  table th {
    padding: 8px 12px;
    vertical-align: top;
    border: 1px solid ${Color.Grey300};
    border-width: 0 0 1px;
  }

  table th {
    font-weight: 700;
  }

  table thead td,
  table thead th {
    border-width: 0 0 2px;
  }

  table tfoot td,
  table tfoot th {
    border-width: 2px 0 0;
  }

  table tbody tr:last-child td,
  table tbody tr:last-child th {
    border-bottom-width: 0;
  }

  ${withMargin};
`;

export default Content;
