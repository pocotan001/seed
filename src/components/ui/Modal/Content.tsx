import styled from "styled-components";
import { colors } from "~/components/styles/theme";

interface IContentProps {
  padded?: boolean;
  actived?: boolean;
}

const Content = styled<IContentProps, "div">("div")`
  position: relative;
  overflow: auto;
  width: 640px;
  ${({ padded }) => padded && "padding: 24px"};
  max-width: calc(100% - 48px);
  max-height: calc(100% - 48px);
  background: ${colors.white};
  border-radius: 5px;
  opacity: ${({ actived }) => (actived ? "1" : "0")};
  transform: ${({ actived }) => `scale(${actived ? "1" : "0.85"})`};
  transition: opacity 0.2s, transform 0.2s;
  -webkit-overflow-scrolling: touch;
`;

export default Content;
