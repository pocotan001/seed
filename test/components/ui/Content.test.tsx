import * as React from "react";
import * as renderer from "react-test-renderer";
import theme from "~/components/styles/theme";
import Content from "~/components/ui/Content";

describe("<Content>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Content theme={theme}>alo</Content>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
