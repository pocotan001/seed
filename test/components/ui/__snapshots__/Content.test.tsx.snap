// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content> renders correctly 1`] = `
<div
  className="sc-bdVaJa eDPbPD"
>
  alo
</div>
`;
